examples/reference/publication_quality/bokeh/bar.py

```diff
@@ -0,0 +1,209 @@
+        line_color=None,            # Remove bar borders
+    ),
+    hv.opts.Labels(
+        text_baseline='bottom',
+        text_font_size='11pt',
+        text_font_style='normal',
+        text_color='#333333',
+    ),
+)
+hv.Cycle.default_cycles["default_colors"] = [ACCENT_COLOR, '#00948A', '#7E59BD', '#FFA20C', '#DA4341', '#D6F1FF', '#DAF5F4', '#F0E8FF', '#FFF8EA', '#FFF1EA', '#001142', '#003336', '#290031', '#371F00', '#3A0C13']
+
+# ============================================================================
+# DATA PIPELINE - Separate extraction, transformation, and plotting
+# ============================================================================
+
+def get_earthquake_data():
+    """Extract raw earthquake data from sample dataset"""
+    return hvsampledata.earthquakes("pandas")
+
+
+def aggregate_by_magnitude(earthquake_data):
+    """Transform: Group earthquakes by magnitude class with statistics"""
+
+    # Aggregate: count events and calculate average depth per magnitude class
+    aggregated = (
+        earthquake_data
+        .groupby('mag_class', observed=True)
+        .agg({'mag': 'count', 'depth': 'mean'})
+        .reset_index()
+        .rename(columns={'mag': 'event_count', 'depth': 'avg_depth'})
+        .sort_values('event_count', ascending=False)
+    )
+
+    # Add percentage column for tooltips
+    aggregated['percentage'] = (
+        aggregated['event_count'] / aggregated['event_count'].sum() * 100
+    )
+
+    return aggregated
+
+
+def create_bar_chart(aggregated_data):
+    """Create publication-quality bar chart with labels and tooltips"""
+
+    default_tools=['save']
+
+    # Main bar chart with professional styling
+    bar_chart = hv.Bars(aggregated_data, kdims='mag_class', vdims=['event_count', 'percentage', 'avg_depth']).opts(
+        # Titles and labels
+        title='Earthquake Distribution by Magnitude',
+        xlabel='Magnitude',
+        ylabel='Number of Events',
+
+        # Interactivity
+        # hover_cols = ["mag_class", "event_count", "percentage", "avg_depth"],
+        hover_tooltips=[
+            ('Magnitude', '@mag_class'),
+            ('Events', '@event_count{0,0}'),      # Format: 1,234
+            ('Percentage', '@percentage{0 a}%'), # Format: 45.7%
+            ('Avg Depth', '@avg_depth{0f} km')  # Format: 99 km
+        ],
+        default_tools=default_tools
+    )
+
+    # Add text labels above bars
+    labels_data = aggregated_data.copy()
+    labels_data['label_y'] = labels_data['event_count'] + 20  # Offset above bars
+
+    text_labels = hv.Labels(labels_data, kdims=['mag_class', 'label_y'], vdims=['event_count', 'percentage', 'avg_depth']).opts(
+        hover_tooltips=[
+            ('Magnitude', '@mag_class'),
+            ('Events', '@event_count{0,0}'),      # Format: 1,234
+            # tooltips below do currently not work on Labels
+            # ('Percentage', '@percentage{0 a}%'), # Format: 45.7%
```

Comment:
Could not get additional tooltips working for Labels. Believe its a bug. Reported in https://github.com/holoviz/holoviews/issues/6769.